### PR TITLE
Check for line ending in output messages in ex_unit_tests.exs

### PR DESCRIPTION
**Pull request by eksperimental**
... and beginning of line when applicable.

This makes sure we check for the full line,
avoiding unexpected results appearing at the end of the line.

Related discussion:
https://github.com/elixir-lang/elixir/pull/8312#discussion_r226867015

### CODE REVIEW

File: lib/ex_unit/test/ex_unit_test.exs [hunk 1]
```diff
@@ -760,15 +762,27 @@ defmodule ExUnitTest do
 
   ##  Helpers
 
-  defp on_exit_reload_config(extra \\ []) do
+  defp on_exit_reload_config() do
```
Comment (josevalim) [inline lib/ex_unit/test/ex_unit_test.exs]: Why are we breaking this in two functions?

Comment (eksperimental) [reply lib/ex_unit/test/ex_unit_test.exs]: the original function was passing extra options, but those were set on_exit. But that extra argument was never used (I don't know if it became unused with the latest changes).

The new function uses for the current module and restores the original configuration on exit.
We could stick with the new one, and replace `on_exit_reload_config/2` calls

Comment (josevalim) [reply lib/ex_unit/test/ex_unit_test.exs]: Good catch. Yes, we should have just one function then. :)

Comment (eksperimental) [reply lib/ex_unit/test/ex_unit_test.exs]: done

**Pull request merged**